llvm/lib/Transforms/InstCombine/InstCombineAtomicRMW.cpp

```diff
@@ -41,6 +41,8 @@ bool isIdempotentRMW(AtomicRMWInst& RMWI) {
     case AtomicRMWInst::Sub:
     case AtomicRMWInst::Or:
     case AtomicRMWInst::Xor:
+    case AtomicRMWInst::USubCond:
+    case AtomicRMWInst::USubSat:
```

Comment:
Missing tests for this (I would also split the instcombine piece into a separate change)

Reply:
I've removed this piece for now. I'm happy to add it back with a separate change, but it looks from the tests like we don't actually want to use idempotency optimizations with atomicrmw operations at all so maybe nothing needs to be done here.

Reply:
The atomicrmw -> atomicrmw idempotent case handled here is fine, it's the atomicrmw -> load atomic case that's problematic in the lowering.

Also in a follow up, the AMDGPU lowering should have the idempotent case to add 0 since that works over PCIe